File: Loenn/entities/strawberry_jam/bullet_time_controller.lua
local bulletTimeController = {}

bulletTimeController.name = "CommunalHelper/SJ/BulletTimeController"
bulletTimeController.depth = -1000000
bulletTimeController.texture = "objects/CommunalHelper/strawberryJam/bulletTimeController/icon"
bulletTimeController.fieldInformation = {
    timerate = {
        minimumValue = 0.01
    },
    minDashes = {
        fieldType = "integer",
        minimumValue = 0
    }
}

bulletTimeController.placements = {
    {
        name = "controller",
        data = {
            timerate = 0.5,
            flag = "",
            minDashes = 1
        }
    }
}

return bulletTimeController

File: Ahorn/entities/strawberryJam/BulletTimeController.jl
module CommunalHelperSJBulletTimeController

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/SJ/BulletTimeController" BTController(x::Integer, y::Integer, timerate::Number = 0.5, flag::String="", minDashes::Integer)

const placements = Ahorn.PlacementDict(
    "Bullet Time Controller (Strawberry Jam) (Communal Helper)" => Ahorn.EntityPlacement(
        BTController,
        "rectangle"
    )
)

function Ahorn.selection(entity::BTController)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x, y, 8, 8)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::BTController, room::Maple.Room)
    Ahorn.drawRectangle(ctx, 0, 0, 8, 8, Ahorn.defaultWhiteColor, Ahorn.defaultBlackColor)
end


end